### ISSUE TITLE
'Container is Falsy'

### ISSUE THREAD

**Issue by bradharms** [WarnerHooh/babel-plugin-parameter-decorator#12]
I am using this plugin in conjunction with Webpack, Happypack, @babel/preset-tyepscript, and Loopback 4. I'm not sure if that makes any difference.

Every file in my project that uses parameter decorators reports "Container is falsy" during compilation. My files look like this:

```typescript
import {inject, CoreBindings} from '@loopback/core';
import { RestApplication } from '@loopback/rest';

export default class MyComponent {
    constructor(
        @inject(CoreBindings.APPLICATION_INSTANCE) app: RestApplication
    ) {
        app.bind(...);
    }
}
```

The error output looks like this (sanitized to hide sensitive project info):

```
ERROR in <PROJECT_ROOT>/src/component.ts
    Module build failed (from ./node_modules/happypack/loader.js):
    ReferenceError: <PROJECT_ROOT>/src/component.ts: Container is falsy
        at NodePath._replaceWith (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/replacement.js:166:11)
        at NodePath.replaceWith (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/replacement.js:156:8)
        at <PROJECT_ROOT>/node_modules/babel-plugin-parameter-decorator/lib/index.js:170:19
        at Array.forEach (<anonymous>)
        at PluginPass.Function (<PROJECT_ROOT>/node_modules/babel-plugin-parameter-decorator/lib/index.js:141:44)
        at newFn (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/visitors.js:179:21)
        at NodePath._call (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/context.js:53:20)
        at NodePath.call (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/context.js:40:17)
        at NodePath.visit (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/context.js:88:12)
        at TraversalContext.visitQueue (<PROJECT_ROOT>/node_modules/@babel/core/node_modules/@babel/traverse/lib/context.js:112:16)
     ...import stack...
```

Here is the part of the webpack config related to transpiling:

```
plugins: [
        // Transpiling of TypeScript is deferred to this plugin rather than
        // going through webpack directly. It allows many files to be processed
        // in parallel, and allows typechecking to be performed in a separate
        // thread from transpiling.
        // On web, TypeScript is transpiled using Babel, because the
        // React Hot Loader plugin won't work with the standard
        // TypeScript loader. (Otherwise we would use ts-loader.)
        new HappyPack({
            id: 'ts',
            loaders: [{
                path: 'babel-loader',
                query: {
                    babelrc: false,
                    cacheDirectory: true,
                    plugins: [
                        'babel-plugin-transform-builtin-extend',
                        [
                            '@babel/plugin-proposal-decorators',
                            { legacy: true },
                        ],
                        [
                            '@babel/plugin-proposal-class-properties',
                            { loose: true },
                        ],
                        'babel-plugin-parameter-decorator',
                        // This allows React components (which are properly
                        // marked) to reload themselves in response to
                        // changes in their source code
                        'react-hot-loader/babel',
                    ],
                    presets: [
                        [
                            '@babel/preset-env',
                            { targets: { /** @todo This is the wrong target string! */ browsers: 'last 2 versions' } },
                        ],
                        '@babel/preset-typescript',
                        '@babel/preset-react',
                    ],
                },
            }],
            threads: 3,
        }),

        // Babel-Typescript can't do type-checking by itself, so we have to use
        // this plugin to do that instead. A pleasant side effect of this is
        // that type checking happens in multiple threads independent of the
        // main build process, which improves build time, and also gives the
        // option of using TS Lint to lint all code if wanted.
        new ForkTsCheckerWebpackPlugin({
            checkSyntacticErrors: true,
            memoryLimit: 2048,
            tsconfig: path.resolve({
                node: NODE_SRC_DIR,
                web: WEB_SRC_DIR,
            }[target], 'tsconfig.json'),
            ...(TSLINT_FIX ? {tslint: true} : {}),
            tslintAutoFix: TSLINT_FIX,
            useTypescriptIncrementalApi: false,
            workers: 2,
        }),
```

**Comment by WarnerHooh** [comment]
Seems the import from `loopback` is failed.

**Comment by bradharms** [comment]
What would cause that to happen? Is there a way to work around it?

**Comment by WarnerHooh** [comment]
I have found this problem, and will fix it soon.
This happened when your class has only the constructor method, without any other method. So a quick temporary workaround is, add some other method in your class.

**Comment by WarnerHooh** [comment]
@bradharms Thanks for your feedback.
Just released a new version `1.0.10` which should fix this issue. Please let me know if there's any problem.

**Comment by bradharms** [comment]
The fix appears to have resolved the build problems. Thank you.
